api/envoy/type/matcher/v3/string.proto

```diff
@@ -53,6 +53,11 @@ message StringMatcher {
     // The input string must match the regular expression specified here.
     RegexMatcher safe_regex = 5 [(validate.rules).message = {required: true}];
   }
+
+  // If true, indicates the exact/prefix/suffix matching should be case insensitive. This has no
+  // effect for the safe_regex match.
```

Comment:
Can you check this at config load and fail if set to true for the regex fields?

Reply:
I added PANIC to the constructor of StringMatcher if ignore_case is used with regex. Please let me know if you mean something else when you say `config load`. I'm not sure how to check this in filter initializing stage because the StringMatcher is used by many filters and I don't know if we have a central place to do the check.

Reply:
Updated to use EnvoyException instead of PANIC.